interview-questions.md

```diff
@@ -14,44 +14,44 @@ Researched answer:
 
-Your answer:
+Your answer: The index of an array describes the location of a given value within that array. The value of an array is the actual component that occupies a given index within an array.
 
-Researched answer:
+Researched answer: Each item within an array is referred to as an element. An array can contain many elements of many different datatypes. An index is akin to the address within an array. It can be used to determine the location of a give value within a particular array.
 
 4. Compare and contrast arrays and strings. Please include ways in which they are the same and ways in which they are different.
 
-Your answer:
+Your answer: Arrays are much like a list. They are ordered in a particular manner and can store many different datatypes within. A string is a collection of characters that can be arranged in any sort of manner. Both strings and arrays have indices which can be used to reference a specific location within the string or array. The essential difference, however, is that an array can contain any number of datatypes, where as a string may only store characters.
 
-Researched answer:
+Researched answer: An array is a data structure that is made up of a collection of elements identified by the index. However, a string is a collection of specifically characters. While an index can be utilize to identify a specific character within a string, it is only capable of containing characters, while an array can contain other datatypes.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Proper pair coding involves two people working on the same program or project. They pair consists of a driver and a navigator. At any given time, only one person can occupy the position of the driver or navigator. The navigator conducts the research and directs the driver in the means of typing out the code. The driver's job is to receive the direction of the navigator and type out the code as instructed. While it is imperative that no one attempts to fulfill two roles at once, it is still essential that they work together to identify any potential issues in the program.
 
-Researched answer:
+Researched answer: Paired programming includes two programmers working on the same program collaboratively. One programmer is the driver, whose job it is to receive direction and type out the code, while the other is the navigator, whose job it is to direct the driver in the aspects of actually typing the code. The number one practice for paired programming is consistent and effective communication between the programmers. It is essential that this be established quickly and effeciently. Secondly, the pair should switch roles between the driver and navigator routinely. This helps to ensure that neither programmer becomes burnt out or overwhelmed. It is also important that the pair submit code frequently. Doing so ensures that few aspects of the program are lost.
 
```

Comment:
Great interview questions! Love that you are elaborating and making sure you are relaying the full extent of your knowledge. Well done, keep it up!